cva6/tests/custom/sv32/vm_dirty_bit_01.S

```diff
@@ -0,0 +1,151 @@
+#=======================================================================
+#      Write Access of Level1 PTE in User and Supervisor mode when dirty
+#      bit is low  
+#-----------------------------------------------------------------------
+# Test Description:
+# If implementation does not sets the pte.D when PTE is written,
+# and PTE has pte.D=0, then attempting to store on that PTE would
+# raise Store/AMO page fault exception, this test
+# covers the following scenarios in both supervisor and user privilege
+# modes for level1 PTE.
+#                           Level 1
+# - Set PTE.D = 0 and test the write access in Supervisor Mode
+# - Set PTE.D = 0 and test the write access in User Mode
+#
+#=======================================================================
+
+#include "macros.h"
+
+#ifdef smode
+    #define SET_PTE_U 0
+#else
+    #define SET_PTE_U PTE_U
+#endif
+
+#define _MMODE_  "M"
+#define _SUMODE_ "SU"
+
+.text
+.global _start
+
+_start:
+    ALL_MEM_PMP                                                            # PMP permission to all the memory
+    la t1,trap_handler                                                     # loads the address of trap handler 
+    csrw mtvec,t1                                                          # sets the mtvec to trap handler 
+    
+# ----------------LEVEL 1 PTE Setup for load and store test------------
+
+    la a1,vm_en                                                            # loads the address of label vm_en
+    mv a0, a1                                                              # set the VA to PA (identity mapping)
+    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_X | PTE_V )              # sets the permission bits
+    PTE_SETUP_RV32 ( a1, a2, t1, a0, pgtb_l1, LEVEL1 )                     # setup the PTE for level1
+ 
+    la a1,rvtest_data                                                      # loads the address of label rvtest_data
+    mv a0, a1                                                              # set the VA to PA (identity mapping)
+    ori a2, x0, ( PTE_V | PTE_A | SET_PTE_U | PTE_X | PTE_W | PTE_R )      # sets the permission bits --No dirty bit
```

Comment:
Sir, I have given all the permissions to validate that there is a fault because of the dirty bit and it does not matter which combination for the PTE is chosen.

Reply:
@MuhammadHammad001 it does not matter at all if you remove the `PTE_X` permission because it is a permission for the label `rvtest_data`. Here, we are only loading and storing data; we are not executing any instructions at that label. Therefore, there is no need for the `PTE_X` permission. You can remove this."